### PR TITLE
Fix duplicate versions for same chart

### PR DESCRIPTION
**Pull request by ninjadq**
* The detailed issue is described in #450
* And there is a PR #454 fixed one scenario of this issue
* But there is another occasion in which users upload chart with prov
* in this PR is to handle this situation in a way similar to #454

Signed-off-by: DQ <dengq@vmware.com>

### CODE REVIEW

**Comment by scbizu** [comment]
Hi @ninjadq , could you add some test cases for this PR ?

**Comment by ninjadq** [comment]
hi @scbizu , added a case similar to chart only scenario

**Review by scbizu — CHANGES_REQUESTED** [review]

**Comment by scbizu** [comment]
Added some review suggestions , thanks for your PR again

File: pkg/chartmuseum/server/multitenant/handlers.go [hunk 1]
```diff
@@ -384,13 +395,19 @@ func (server *MultiTenantServer) getChartAndProvFiles(req *http.Request, repo st
 		if _, ok := cpFiles[filename]; ok {
 			continue
 		}
-		if status, err := server.validateChartOrProv(repo, filename, force); err != nil {
+		status, err := server.validateChartOrProv(repo, filename, force)
```
[the remark this answers is not in the document]
Comment (ninjadq) [reply pkg/chartmuseum/server/multitenant/handlers.go:404]: hi @scbizu,  The `server.validateChartOrProv` function not only check the existence of files but also check if their filename is improperly formatted. I just wonder if we can ignore this filename format check? If we can I will update this.

[the remark this answers is not in the document]
Comment (scbizu) [reply pkg/chartmuseum/server/multitenant/handlers.go:404]: Maybe we can move the filename validation out of `validateChartOrProv`  and put it to this loop  before `exist` checking? Like :

```Go

// check filename 

// existence checking
if exist { 
  cpFiles[filename] = &chartOrProvenanceFile{filename, content, ff.field} 
  continue 
}
// ... 

```

( BTW , the filename is constructed internally , I do not find any risks if we remove/ignore this validation)

[the remark this answers is not in the document]
Comment (ninjadq) [reply pkg/chartmuseum/server/multitenant/handlers.go:414]: hi @scbizu, thank you for your reply, I've updated the code and move the format check out of the `validateChartOrProv` func.

[the remark this answers is not in the document]
Comment (scbizu) [reply pkg/chartmuseum/server/multitenant/handlers.go:414]: Thank you XD

**Review by scbizu — APPROVED** [review]

**Comment by scbizu** [comment]
LGTM 😎

**Comment by scbizu** [comment]
Merging , the bad CI will be fixed in #482

**Comment by ninjadq** [comment]
Thank you @scbizu, happy holidays

**Comment by scbizu** [comment]
@ninjadq You too

**Pull request merged**